src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/UpdateTribunalDecisionHandler.java

```diff
@@ -95,14 +97,29 @@ public PreSubmitCallbackResponse<AsylumCase> handle(PreSubmitCallbackStage callb
                     decisionAndReasonsAppender.append(newDecisionAndReasons, maybeExistingDecisionAndReasons.orElse(emptyList()));
 
             asylumCase.write(CORRECTED_DECISION_AND_REASONS, allCorrectedDecisions);
+
             final Optional<Document> decisionAndReasonsDoc = asylumCase.read(DECISION_AND_REASON_DOCS_UPLOAD, Document.class);
 
+            if (decisionAndReasonsDoc.isPresent()) {
+                DocumentWithMetadata updatedDecisionAndReasonsDocument = documentReceiver.receive(
+                    decisionAndReasonsDoc.get(),
+                    "",
+                    DocumentTag.UPDATED_FINAL_DECISION_AND_REASONS_PDF
+                );
+
+                List<IdValue<DocumentWithMetadata>> newUpdateTribunalDecisionDocs = documentsAppender.append(
+                    new ArrayList<>(),
```

Comment:
Append to a new list wouldn't that wipe out all the decisions and reasons document? (cover letters, original decision document, and AMENDED cover letter)